### ISSUE TITLE
Allow running something else than state.highstate

### ISSUE THREAD

**Issue by avanier** [saltstack/kitchen-salt#202]
It would be handy to able to test only a specific state.

I have an idea of how to do this, so PR incoming at some point.

The work starts [here-ish~](https://github.com/saltstack/kitchen-salt/blob/master/lib/kitchen/provisioner/salt_solo.rb#L234).

```python
 cmd << sudo("#{salt_call} --state-output=changes --config-dir=#{File.join(config[:root_path], salt_config_path)} --local state.highstate") # <--- Worky, worky...
```

**Comment by gtmanfred** [comment]
probably best to just do a state.apply for both, and then populate a list called `sls` that gets added if something is actually there.

The problem with not using a highstate, is any modules may not be synced, i can't remember if dynamic modules are synced on the minion starting up, but i know they aren't synced for state.sls, so I would like to see an option added around syncing all, (since refresh def happens with salt-call), and then also add a test to the suite to test that feature.

**Comment by avanier** [comment]
I keep making those discoveries about Salt. I'm really a Chef user stuck in a Salt world right now.

Thanks for the heads-up / plan. I'll dig and report.

**Comment by avanier** [comment]
Thanks again for this info, this just saved me a lot of trouble for something else altogether. Also, part of the answers we seek are [here](https://docs.saltstack.com/en/latest/ref/modules/all/salt.modules.saltutil.html#salt.modules.saltutil.sync_all).

**Comment by gtmanfred** [comment]
correct, you could optionally just have a `sync:` section that people could put in the different things to sync, like `grains` or `modules` or potentially just set it to `true` or all and it will do that.

**Comment by zachsmorgan** [comment]
I think it would be worth having a configuration option to override the salt-apply command / args.

See https://github.com/saltstack/kitchen-salt/pull/236

**Comment by zachsmorgan** [comment]
Just to bring this full circle, I ended up switching to the shell provisioner. I don't think I had thought the ramifications for my PR through before submitting, so, sorry for the extra noise. :)
